department-of-veterans-affairs/vets-website
# Add unauthenticated warning alert

## ⚠️ CAUTION: DO NOT MERGE THIS PR UNTIL 'PENSION_FORM_ENABLED' IS DISABLED IN PRODUCTION

## Summary
On the Pension Benefits application(21P-527EZ),
On the Step 1 Applicant Information page

Add unauthenticated warning alert at top of page

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
Unauthenticated
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Select 'Start your application without signing in'
3. On Step 1 - Applicant Information verify the unauthenticated warning alert is visible
4. Verify that the 'sign in to save progress' link opens the sign in modal

Authenticated 
1. Sign in via `http://localhost:3001/sign-in/mocked-auth` to authenticate your user
2. Go to `http://localhost:3001/pension/application/527EZ` in your browser
3. Select Start application or continue application
4. On Step 1 - Applicant Information verify the unauthenticated warning alert is not visible

## Related issue(s)
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68557)
[Epic in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)

## Testing done

- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots
_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

### Unauthenticated Mobile
| Before | After |
| ------ | ----- |
|![localhost_3001_pension_application_527EZ_applicant_information(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/245117bf-a62a-4e7b-8701-a31d5722f69e)|![localhost_3001_pension_application_527EZ_(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/eabaf306-76cb-4c5f-830e-be23f8225336)


### Unauthenticated Desktop
| Before | After |
| ------ | ----- |
|![localhost_3001_pension_application_527EZ_applicant_information](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/c14443e1-78f5-4f6c-8297-cdd89a00d009)|![localhost_3001_pension_application_527EZ_applicant_information_unauth](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/90c0c684-0d9d-4140-a9a2-7d3b98c23b37)


## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
- On the application forms fist chapter and page, there will be a new field in between the Social Security Number field and the Date of Birth field
- It will be labeled “Have you filed any type of VA claim before?” with options for “Yes” and “No.”
- If the user selects “Yes,” an additional field labeled “VA File Number” will appear.
- The form will allow for a Veteran whose file number is different than their SSN to enter both. This field will be optional for the user.
- The “VA File Number” field should have validation to ensure it follows the correct format.
- The field responses to this question should be saved so that it can be submitted.

### Quality Assurance & Testing

### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
